python_utils.py

```diff
@@ -399,6 +399,12 @@ def recursively_convert_to_str(value):
             recursively_convert_to_str(k): recursively_convert_to_str(
                 v) for k, v in value.items()}
     elif isinstance(value, BASESTRING):
-        return str(value)
+        # We are importing utils here to prevent circular imports.
+        import utils
+        temp = str(utils.to_ascii(value))
+        if temp.startswith('b\''):
```

Comment:
When we wrap str around a builtins.bytes object, the b' prefix will get embedded. I wanted to do this so that I can differentiate between builtins and normal str. This is necessary as comparing types doesn't work as python treats builtins.str and unicode the same

Reply:
Actually wrapping in str is necessary. This is the only way we can convert future.types.xxx to normal str/bytes

Reply:
Added a comment for more details!